bug: test_get_cost_summary_daily_breakdown uses hardcoded dates that fall outside 7d window

## Bug

`tests/test_cost_routes.py::test_get_cost_summary_daily_breakdown` is failing because it creates sessions with hardcoded dates `2026-04-01` and `2026-04-02`, which are now more than 7 days in the past. The `/api/cost/summary?range=7d` query filters them out, so `daily_sum` returns `0.0` against the expected `>= 5.25`.

## Root Cause

The test hardcodes `started_at=datetime(2026, 4, 1, ...)` and `datetime(2026, 4, 2, ...)` instead of using dates relative to `datetime.now(UTC)`.

## Fix

Replace hardcoded dates with `datetime.now(timezone.utc) - timedelta(days=N)` so the sessions always fall within the 7-day window regardless of when the test runs.

## Failing assertion

```
assert daily_sum >= expected_daily_total
AssertionError: assert 0.0 >= 5.25
```

--- a/tests/test_cost_routes.py
+++ b/tests/test_cost_routes.py
@@ -2,7 +2,7 @@
 
 from __future__ import annotations
 
-from datetime import datetime, timezone
+from datetime import datetime, timedelta, timezone
 
 import pytest
 from httpx import ASGITransport, AsyncClient
@@ -240,7 +240,7 @@ async def test_get_cost_summary_with_range():
                 _make_session(
                     "cost-range-recent",
                     total_cost_usd=2.00,
-                    started_at=datetime(2026, 4, 1, 10, 0, tzinfo=timezone.utc),
+                    started_at=datetime.now(timezone.utc) - timedelta(days=3),
                 )
             )
             # Old session (more than 7 days ago)
@@ -248,7 +248,7 @@ async def test_get_cost_summary_with_range():
                 _make_session(
                     "cost-range-old",
                     total_cost_usd=5.00,
-                    started_at=datetime(2026, 3, 20, 10, 0, tzinfo=timezone.utc),
+                    started_at=datetime.now(timezone.utc) - timedelta(days=30),
                 )
             )
             await db_session.commit()
@@ -276,25 +276,26 @@ async def test_get_cost_summary_daily_breakdown():
         # Create sessions across multiple days
         async with app_context.require_session_maker()() as db_session:
             repo = TraceRepository(db_session)
+            _now = datetime.now(timezone.utc)
             await repo.create_session(
                 _make_session(
                     "cost-daily-1",
                     total_cost_usd=1.00,
-                    started_at=datetime(2026, 4, 1, 10, 0, tzinfo=timezone.utc),
+                    started_at=_now - timedelta(days=3),
                 )
             )
             await repo.create_session(
                 _make_session(
                     "cost-daily-2",
                     total_cost_usd=2.50,
-                    started_at=datetime(2026, 4, 2, 10, 0, tzinfo=timezone.utc),
+                    started_at=_now - timedelta(days=2),
                 )
             )
             await repo.create_session(
                 _make_session(
                     "cost-daily-3",
                     total_cost_usd=1.75,
-                    started_at=datetime(2026, 4, 2, 15, 0, tzinfo=timezone.utc),
+                    started_at=(_now - timedelta(days=2)).replace(hour=15),
                 )
             )
             await db_session.commit()

--- a/tests/test_cost_routes.py
+++ b/tests/test_cost_routes.py
@@ -27,7 +27,7 @@ def _make_session(
         agent_name="test_agent",
         framework=framework,
         started_at=_started,
-        ended_at=_started.replace(hour=11),
+        ended_at=_started + timedelta(hours=1),
         status=SessionStatus.COMPLETED,
         total_cost_usd=total_cost_usd,
         total_tokens=total_tokens,

